docs/deprecation.md

```diff
@@ -39,5 +39,55 @@ In order to deprecate a feature gate the following conditions must be met:
 
 After the above deprecation process, the feature gate should remain as a no-op with a warning popping up.
 
+### Metadata (e.g. labels / annotations)
+Kubevirt adds certain labels / annotations to both objects that is created by us (e.g. VMs) and cluster-wide
+objects (e.g. Nodes). This metadata is exposed to the end-user which might use it and depend on it.
+
+Usually, metadata needs to be deprecated if it has a non-intuitive / irrelevant name or reflects data that is no
+longer relevant / useful.
+
+#### Deprecation
+In order to deprecate metadata the following conditions must be met:
+* A mail should be sent to kubevirt-dev mailing list (kubevirt-dev@googlegroups.com) to inform about the deprecation.
+* In case the metadata is being renamed / changed, the new metadata (e.g. label) should be added alongside the deprecated
+one.
+
+For example, if we want to deprecate a label named `kubevirt.io/badNonIntuitiveName` to
+`kubevirt.io/greatNewName`, both labels should appear.
+
+#### Removal
+Deprecated metadata will not be removed for at least 3 release cycles, which means roughly 1 year.
+
+Afterwards, a mail should be sent to kubevirt-dev mailing list (kubevirt-dev@googlegroups.com) to inform
+about the removal. If there are no objections, the metadata can be removed. Otherwise, a new removal date
+can be discussed according to the context.
+
+### API
+Although the following only mentions API objects, it's also relevant for their APIs (e.g. object fields),
+Kubevirt CR configuration, etc. Obviously, for fields / configs deprecation the step about adding a
+deprecated annotations can be skipped. However, a warning should always be triggered for any usage of all
+of the mentioned above.
+
+#### Deprecation
+In order to deprecate an API object the following actions need to be taken:
+* The deprecated object needs to contain a warning about the object being deprecated. The warning should also
+  state whenever the object would be removed (see below).
+* A `"kubevirt.io/deprecated":"true"` annotation should be added to all of the object instances.
+* A mail should be sent to kubevirt-dev mailing list (kubevirt-dev@googlegroups.com) to inform about the deprecation.
+
+#### Removal
+A deprecated object will not be removed for at least 5 release cycles, which means roughly 1.5 years.
+
+Afterwards, a mail should be sent to kubevirt-dev mailing list (kubevirt-dev@googlegroups.com) to inform
+about the removal. If there are no objections, the object can be removed. Otherwise, a new removal date
+can be discussed according to the context.
+
+### Feature Behavior
```

Comment:
Maybe the API section could just say "API / Feature Behavior"? And then the removal time could say "X time (Y for Feature Behavior)". I feel it would be a bit clearer, but up to you!
Also it would be nice to have an example of some feature behavior that falls into this category, and some that don't.

Reply:
Hey! :)

> Maybe the API section could just say "API / Feature Behavior"? And then the removal time could say "X time (Y for Feature Behavior)". I feel it would be a bit clearer, but up to you!

The reason I prefer to separate two the is that this PR only sets the foundation for a deprecation policy. I expect that this document will grow and change as time goes. Today API and feature behavior changes are almost identical, but that can change in the future.

> Also it would be nice to have an example of some feature behavior that falls into this category, and some that don't.

Great suggestion! Added!